app/com/arpnetworking/metrics/portal/reports/ReportRepository.java

```diff
@@ -43,6 +43,16 @@
      */
     Optional<Report> getReport(UUID identifier, Organization organization);
 
+    /**
+     * Delete a {@code Report} given its identifier.
+     *
+     * @param identifier The {@code Report} identifier.
+     * @param organization The {@code Organization} which owns the report.
+     * @throws javax.persistence.PersistenceException if the delete was unsuccessful.
+     * @return the number of reports deleted. Should be 0 or 1.
+     */
+    int deleteReport(UUID identifier, Organization organization);
```

Comment:
Returning an int isn't the most intuitive thing IMO, I would rather this throw `EntityNotFound` or return an Optional (maybe containing the deleted report?) if unsuccessful. That seems like it would be harder to mess up in implementation / error handling. I only did it like this since I rank consistency with the existing repositories higher in my considerations.

Reply:
First, you probably shouldn't throw `javax.persistence.PersistenceException` since that seems implementation specific. 

Second, I totally agree with you that this interface definition is borked. I would prefer to see it return the deleted entity although if that's not possible just return nothing and throw on failure. However, we would need to apply that refactoring all the `delete` repository interfaces for consistency. If you want to do that here, I'm fine with it, otherwise, we can file an issue and refactor it later.

Reply:
Opinion: Runtime exceptions should not be declared thrown on the signature or on the Javadoc because you should not take logical action based on runtime exceptions. There is unfortunately a lot of misclassification of exceptions in the Java environment, where handle-able exceptions are often typed as runtime exceptions because it is perceived as giving programmers more choice.

In the case of this project, the repository interfaces throw exceptions but only those they don't expect clients to handle. For example, delete should not throw if the item to be deleted does not exist (it should return 0). Delete should throw a runtime exception subclass if the delete itself fails.

There is a perfectly reasonable argument to be made that the repositories should throw checked exceptions on failure to allow clients to handle and recover appropriately. I'm not against that, we just don't have a use case for it in the project and as such it was punted because designing the right granularity of exceptions across all repository interfaces and implementations was deemed to have low ROI.

Happy to discuss this more.